common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,36 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def is_root_block(xblock, is_root=False):
```

Comment:
My concern would be around performance. If walking to the root is fast, then I like the idea of filtering parents to ignore the ones in orphaned chains.